packages/kbn-interpreter/src/server/get_plugin_paths.js

```diff
@@ -26,80 +26,39 @@ import { pluginPaths } from './plugin_paths';
 const lstat = promisify(fs.lstat);
 const readdir = promisify(fs.readdir);
 
-const canvasPluginDirectoryName = 'canvas_plugin';
-
 const isDirectory = path =>
   lstat(path)
     .then(stat => stat.isDirectory())
     .catch(() => false); // if lstat fails, it doesn't exist and is not a directory
 
-const isDirname = (p, name) => path.basename(p) === name;
-
 const filterDirectories = (paths, { exclude = false } = {}) => {
   return Promise.all(paths.map(p => isDirectory(p))).then(directories => {
     return paths.filter((p, i) => (exclude ? !directories[i] : directories[i]));
   });
 };
 
-const getPackagePluginPath = () => {
-  let basePluginPath = path.resolve(__dirname, '..');
-
-  if (isDirname(basePluginPath, 'target')) {
-    basePluginPath = path.join(basePluginPath, '..');
-  }
-  return basePluginPath;
-};
-
-const getKibanaPluginsPath = () => {
-  let kibanaPath = path.resolve(getPackagePluginPath(), '..', '..');
-
-  // in dev mode we are in kibana folder, else we are in node_modules
-  if (!isDirname(kibanaPath, 'kibana')) {
-    kibanaPath = path.join(kibanaPath, '..');
-  }
-
-  return path.join(kibanaPath, 'plugins');
-};
-
-const getXPackPluginsPath = () => {
-  const kibanaPath = path.resolve(getPackagePluginPath(), '..', '..');
-
-  // in dev mode we are in kibana folder, else we are in node_modules
-  return path.join(kibanaPath, 'x-pack/plugins');
-};
-
-// These must all exist
-const paths = [
-  getPackagePluginPath(),
-  getXPackPluginsPath(), // Canvas core plugins
-  getKibanaPluginsPath(), // Kibana plugin directory
-].filter(Boolean);
 
-export const getPluginPaths = type => {
+export const getPluginPaths = (pathsRegistry, type) => {
   const typePath = pluginPaths[type];
   if (!typePath) throw new Error(`Unknown type: ${type}`);
+  const paths = pathsRegistry.get(type);
```

Comment:
`PathsRegistry` class above doesn't seem to take in a `type` parameter into `get`, unless this isn't a `PathsRegistry` object?